homeassistant/components/lifx/light.py

```diff
@@ -408,7 +412,7 @@ async def register_new_bulb(self, bulb):
             color_resp = await ack(bulb.get_color)
 
             if color_resp is None or version_resp is None:
-                _LOGGER.error("Failed to initialize %s", bulb.ip_addr)
+                _LOGGER.error("Failed to connect to %s", bulb.ip_addr)
                 bulb.registered = False
             else:
                 bulb.timeout = MESSAGE_TIMEOUT
```

Comment:
In a future PR we should return above, drop the else, and outdent below

Reply:
I _think_ I get what you're suggesting. Will try it out and see what happens.

Reply:
I'm not entirely sure what the point of this block is any more. Instead of just this minor change, I'd prefer to refactor the whole discovery process as a whole, while moving it out of the light entity as well.

Reply:
I can explain but I am not sure which block you are talking about. Will look at the logs now.

Reply:
@amelchio I've mostly worked it out and it's still somewhat relevant given the current discovery process. I got a single `missing entity_id` error when I first tested @bdraco's suggestion, but I can't seem to repeat that. I'd like to get this merged and tested in the beta though, and will continue to optimise where I can.